lib/c-api/src/wasm_c_api/engine.rs

```diff
@@ -72,6 +74,7 @@ impl Default for wasmer_engine_t {
 #[derive(Debug, Default)]
 #[repr(C)]
 pub struct wasm_config_t {
+    #[cfg(feature = "compiler")]
```

Comment:
Minor comment, but it's a bit unfortunate that if the user's libwasmer and wasmer_wasm.h don't agree on the `compiler` feature, this will likely silently do the wrong thing because it overlaps with `engine` and uses the same numbers.

We can probably ignore this for now, but it'd be nice if there was some kind of way to fail at compile time or link time if the header differs from the library being used...

Actually in the case of building something for shared wasmer, this could actually be a problem...

One hacky fix for now is to move `engine` above `compiler` in `wasm_config_t`... then at least it won't confilct